viewer/src/providers/FilterProvider.tsx

```diff
@@ -132,11 +152,13 @@ export function FilterProvider({ children }: PropsWithChildren) {
                 const hasStateFilter = stateFilter.length !== 0
                 const hasObjectTypeFilter = objectTypesFilter.length !== 0
                 const hasSoftwareComponents = softwareComponentsFilter.length !== 0
+                const hasApplicationComponents = applicationComponentsFilter.length !== 0
                 const hasLabelsFilter = labelsFilter.length !== 0
 
                 return  (hasStateFilter ? stateFilter : Object.keys(states)).includes(value.state as string) &&
```

Comment:
this return statement is quite hard to read; maybe extract each condition into a named constant instead?